routing/router_test.go

```diff
@@ -4223,22 +4228,27 @@ func TestSendMPPaymentFailedWithShardsInFlight(t *testing.T) {
 		failedAttempt = payment.HTLCs[0]
 		failedAttempt.Failure = &channeldb.HTLCFailInfo{}
 		payment.HTLCs[0] = failedAttempt
-	}).Once()
+	})
```

Comment:
In this case we can check that the function has been called 4 times using `.Times(4)`.

Reply:
The reason I removed the exact count match is because shards are launched one by one and sometimes we get an exact count, sometimes we don't since if the failure propagates the new shards won't be launched.